### PR TITLE
NN-4052: urlGenerator to template refactor

### PR DESCRIPTION
**Pull request by natclamp-moj**

### CODE REVIEW

File: integration_tests/integration/home.spec.ts [hunk 1]
```diff
@@ -11,29 +11,29 @@ context('Home page', () => {
...
   it('should only see some tiles without the reviewer role', () => {
-    cy.visit(homepageUrl.root)
-    const homepage: Homepage = Page.verifyOnPage(Homepage)
-    homepage.feedbackBanner().should('exist')
-    homepage.startANewReportLink().should('exist')
-    homepage.continueAReportLink().should('exist')
-    homepage.viewYourCompletedReportsLink().should('exist')
-    homepage.viewAllCompletedReportsLink().should('not.exist')
+    cy.visit(adjudicationUrls.homepage.root)
+    const homepageMockup: HomepagePage = Page.verifyOnPage(HomepagePage)
```
Comment (richardpopple) [inline integration_tests/integration/home.spec.ts:21]: Not that its a big deal - but did you need to change the name from homepage to homepagemock? I'm surprised

File: integration_tests/integration/login.spec.ts [hunk 2]
```diff
@@ -11,21 +11,21 @@ context('Sign in', () => {
...
+    cy.visit(adjudicationUrls.homepage.root)
     Page.verifyOnPage(AuthSignInPage)
   })
 
   it('User name visible in header', () => {
     cy.signIn()
-    const homePage: Homepage = Page.verifyOnPage(Homepage)
-    homePage.signInName().contains('J. Smith')
-    homePage.activeLocation().contains('Moorland')
+    const homePageMockup: HomepagePage = Page.verifyOnPage(HomepagePage)
```
Comment (richardpopple) [inline integration_tests/integration/login.spec.ts:20]: Again - I'm surprised you need to change the name here

File: server/utils/nunjucksSetup.ts [hunk 3]
```diff
@@ -132,18 +122,7 @@ export default function nunjucksSetup(app: express.Express, path: pathModule.Pla
   njkEnv.addGlobal('authUrl', config.apis.hmppsAuth.url)
   njkEnv.addGlobal('digitalPrisonServiceUrl', config.digitalPrisonServiceUrl)
   njkEnv.addGlobal('supportUrl', config.supportUrl)
-  njkEnv.addGlobal('homepageUrl', homepageUrl.root)
+  njkEnv.addGlobal('homepageUrl', adjudicationUrls.homepage.root)
```
Comment (richardpopple) [inline server/utils/nunjucksSetup.ts:125]: Do we need this one at all any more?

File: server/utils/nunjucksSetup.ts [hunk 4]
```diff
@@ -132,18 +122,7 @@ export default function nunjucksSetup(app: express.Express, path: pathModule.Pla
...
-  njkEnv.addFilter('incidentDetailsStart', incidentDetails.urls.start)
-  njkEnv.addFilter('incidentDetailsEdit', incidentDetails.urls.edit)
-  njkEnv.addFilter('incidentStatement', incidentStatementUrls.urls.start)
-  njkEnv.addFilter('incidentStatementSubmittedEdit', incidentStatementUrls.urls.submittedEdit)
-  njkEnv.addFilter('checkYourAnswers', checkYourAnswers.urls.start)
-  njkEnv.addFilter('taskList', taskList.urls.start)
-  njkEnv.addFilter('printPdf', printPdf.urls.start)
-  njkEnv.addFilter('prisonerReportReporterView', prisonerReport.urls.report)
-  njkEnv.addFilter('prisonerReportReviewerView', prisonerReport.urls.review)
+  njkEnv.addGlobal('adjudicationUrls', adjudicationUrls)
```
Comment (richardpopple) [inline server/utils/nunjucksSetup.ts:127]: Nice!

File: integration_tests/integration/home.spec.ts (hunk 1 above)
Comment (natclamp-moj) [reply integration_tests/integration/home.spec.ts:21]: yep good spot, this was done before the bulk of the other changes and now it's unnecessary

**Pull request merged**